libs/blog/authors/ui-author-card/src/lib/author-card/author-card.component.ts

```diff
@@ -48,6 +50,21 @@ export class AuthorCardComponent {
     () => 'text-sm' + (this.clampText() ? ' line-clamp-3' : ''),
   );
 
+  private readonly _authorTitlesTranslations = inject(
+    TranslocoService,
+  ).translateObject('authorCard.titles') as { [titleKey: string]: string };
+  readonly authorTitles = computed(
+    () =>
+      this.author()
+        ?.titles.reduce((titlesList, titleKey) => {
+          // skip titles without translation (e.g. 'hoa')
+          return this._authorTitlesTranslations[titleKey]
+            ? titlesList.concat(this._authorTitlesTranslations[titleKey])
+            : titlesList;
+        }, [] as string[])
+        .join(', ') ?? '',
+  );
```

Comment:
suggestion to not use `reduce` to improve readability
```suggestion
  readonly authorTitles = computed(
    () => author()?.titles
      .filter((titleKey) => this._authorTitlesTranslations[titleKey])
      .map((titleKey) => this._authorTitlesTranslations[titleKey])
      .join(', ') ?? '';
  );
```

or
```ts
  readonly authorTitles = computed(
    () => author()?.titles
      .map((titleKey) => this._authorTitlesTranslations[titleKey])
      .filter(Boolean)
      .join(', ') ?? '';
  );
```

will it work?

Reply:
yep, thanks